examples/todo-list-strict/restapi/operations/todos/update_one_responses.go

```diff
@@ -0,0 +1,140 @@
+const UpdateOneOKCode int = 200
+
+/*UpdateOneOK OK
+
+swagger:response updateOneOK
+*/
+type UpdateOneOK struct {
+
+	/*
+	  In: Body
+	*/
+	Payload *models.Item `json:"body,omitempty"`
+}
+
+// NewUpdateOneOK creates UpdateOneOK with default headers values
+func NewUpdateOneOK() *UpdateOneOK {
+
+	return &UpdateOneOK{}
+}
+
+// WithPayload adds the payload to the update one o k response
+func (o *UpdateOneOK) WithPayload(payload *models.Item) *UpdateOneOK {
+	o.Payload = payload
+	return o
+}
+
+// SetPayload sets the payload to the update one o k response
+func (o *UpdateOneOK) SetPayload(payload *models.Item) {
+	o.Payload = payload
+}
+
+// WriteResponse to the client
+func (o *UpdateOneOK) WriteResponse(rw http.ResponseWriter, producer runtime.Producer) {
+
+	rw.WriteHeader(200)
+	if o.Payload != nil {
+		payload := o.Payload
+		if err := producer.Produce(rw, payload); err != nil {
+			panic(err) // let the recovery middleware deal with this
+		}
+	}
+}
+
+func (o *UpdateOneOK) UpdateOneResponder() {}
+
+/*UpdateOneDefault error
+
+swagger:response updateOneDefault
+*/
+type UpdateOneDefault struct {
+	_statusCode int
+
+	/*
+	  In: Body
+	*/
+	Payload *models.Error `json:"body,omitempty"`
+}
+
+// NewUpdateOneDefault creates UpdateOneDefault with default headers values
+func NewUpdateOneDefault(code int) *UpdateOneDefault {
+	if code <= 0 {
+		code = 500
+	}
+
+	return &UpdateOneDefault{
+		_statusCode: code,
+	}
+}
+
+// WithStatusCode adds the status to the update one default response
+func (o *UpdateOneDefault) WithStatusCode(code int) *UpdateOneDefault {
+	o._statusCode = code
+	return o
+}
+
+// SetStatusCode sets the status to the update one default response
+func (o *UpdateOneDefault) SetStatusCode(code int) {
+	o._statusCode = code
+}
+
+// WithPayload adds the payload to the update one default response
+func (o *UpdateOneDefault) WithPayload(payload *models.Error) *UpdateOneDefault {
+	o.Payload = payload
+	return o
+}
+
+// SetPayload sets the payload to the update one default response
+func (o *UpdateOneDefault) SetPayload(payload *models.Error) {
+	o.Payload = payload
+}
+
+// WriteResponse to the client
+func (o *UpdateOneDefault) WriteResponse(rw http.ResponseWriter, producer runtime.Producer) {
+
+	rw.WriteHeader(o._statusCode)
+	if o.Payload != nil {
+		payload := o.Payload
+		if err := producer.Produce(rw, payload); err != nil {
+			panic(err) // let the recovery middleware deal with this
+		}
+	}
+}
+
+func (o *UpdateOneDefault) UpdateOneResponder() {}
+
+type UpdateOneNotImplementedResponder struct {
+	middleware.Responder
+}
+
+func (*UpdateOneNotImplementedResponder) UpdateOneResponder() {}
+
+func UpdateOneNotImplemented() UpdateOneResponder {
+	return &UpdateOneNotImplementedResponder{
+		middleware.NotImplemented(
+			"operation authentication.UpdateOne has not yet been implemented",
+		),
+	}
+}
+
+type UpdateOneResponder interface {
+	middleware.Responder
+	UpdateOneResponder()
```

Comment:
What is the intent with this? Create a sum type?

Reply:
the main idea is to disallow to return a wrong responder from a handler.

we have special types generated by go-swagger intended to be used to return something from the method, so let's use them

Reply:
Ok in that case it might make sense to use a more unique name and perhaps unexport it (for example prefix it with the package name or snakeized full package path).

Right now if something has the signature UpdateOneResponder as a method it can be used as a responder without any guarantee that it is actually one of the generated ones. 

One more suggestion would be to have some helper to wrap a middleware responder with the updateoneresponder so that people can still provide their own but intentionally

Reply:
> Ok in that case it might make sense to use a more unique name and perhaps unexport it (for example prefix it with the package name or snakeized full package path).

It is unique already, isn' it? I did introduce these types, they were there already. I've just changed the Handler code to accept the limited set of types only as of the responders.

> Right now if something has the signature UpdateOneResponder as a method it can be used as a responder without any guarantee that it is actually one of the generated ones.

I do not think it could happen accidentally, do you? And we are not trying to stop the programmer from doing something intentionally, right?

> One more suggestion would be to have some helper to wrap a middleware responder with the updateoneresponder so that people can still provide their own but intentionally

2 things crossing my mind:

1. We have all the methods returning anything described in the spec and so the client can/will expect to receive. The main idea behind this patch is to stop my colleagues from returning custom answers. It was easier to return something custom rather the generated one, it is not any more so problem solved.
2. It is still easy to create a custom time conforms the requirements, but not too easy